btsieve/src/ethereum/queries.rs

```diff
@@ -0,0 +1,655 @@
+use ethbloom::Input;
+use ethereum_support::{
+    web3::types::{TransactionReceipt, H256},
+    Address, Block, Bytes, Transaction,
+};
+
+#[derive(Clone, Default, Debug, Eq, PartialEq)]
+pub struct TransactionQuery {
+    pub from_address: Option<Address>,
+    pub to_address: Option<Address>,
+    pub is_contract_creation: Option<bool>,
+    pub transaction_data: Option<Bytes>,
+    pub transaction_data_length: Option<usize>,
+    pub events: Vec<Event>, // Empty if we are not matching events.
```

Comment:
Could make a this an `Option` to neglect the need for the comment.

Reply:
I think we considered this when the `EventQuery` was originally introduced, but I still don't like it because then we can have `Some(vec![])`, and what does that express? Isn't that just equivalent to `None`? Seems weird to me.

Reply:
Hmm, `Some(vec![])` would mean you want a transaction that doesn't have any events whereas `None` would mean, I don't want to match against the events of the transaction. Not sure if we need it but it would be more consistent with the rest :)

Reply:
I disagree, I don't feel that `Some(vec![])` adds any meaning over an empty vector.  There is no time, AFAICT, that processing an empty vector is different from `None` - so to me it adds no extra meaning.  Any implied meaning would have to be added as a comment anyways because it is would be unclear as to what is the difference.